File: 991-broken-calculator/991-broken-calculator.py
class Solution:
    def brokenCalc(self, startValue: int, target: int) -> int:
        count = 0
        while target != startValue:
            
            if target < startValue:
                count += startValue - target
                return count
            
            
            elif target + 1 == startValue:
                target += 1
            
            elif target % 2 == 0:
                target //= 2
                
            else:
                target += 1
            count += 1
        
        return count

File: 991-broken-calculator/991-broken-calculator.cpp
class Solution {
public:
    int brokenCalc(int startValue, int target) {
        // return 1;
        return solve(startValue, target);
    }
private:
    int solve(int startValue, int target) {
        int count = 0;
        while(target != startValue) {
            if(target < startValue){
                count += startValue - target;
                return count;
            }
            
            else if(target + 1 == startValue)
                target += 1;
            
            else if(target % 2 == 0)
                target /= 2;
            else
                ++target;
            ++count;
        }
        return count;
    }
};